examples/vault-consul-ami/vault-consul.json

```diff
@@ -57,7 +59,11 @@
   },{
     "type": "shell",
     "inline": [
-      "/tmp/terraform-aws-vault/modules/install-vault/install-vault --version {{user `vault_version`}}"
+      "if [[ -n '{{user `vault_download_url`}}' ]]; then",
+      " /tmp/terraform-aws-vault/modules/install-vault/install-vault --download-url {{user `vault_download_url`}};",
+      "else",
+      " /tmp/terraform-aws-vault/modules/install-vault/install-vault --version {{user `vault_version`}};",
```

Comment:
Instead of this if-statement, another option to consider is that you can set both `--download-url` and `--version`, except one will be set to empty string and ignored.

Reply:
(Carrying this discussion on the other PR)